### PR TITLE
feat(gh-comments): Queuing Logic

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
Add task queuing logic to post process, specifically in process_commit_context when a suspect committer is established.

### CODE REVIEW

File: tests/sentry/tasks/test_commit_context.py [hunk 1]
```diff
@@ -408,3 +411,157 @@ def after_return(self, status, retval, task_id, args, kwargs, einfo):
...
             assert mock_suspect_commits.called
+
+
+@region_silo_test(stable=True)
+@patch(
+    "sentry.integrations.github.GitHubIntegration.get_commit_context",
+    Mock(
+        return_value={
+            "commitId": "asdfwreqr",
+            "committedDate": "2023-02-14T11:11Z",
```
Comment (sentaur-athena) [inline tests/sentry/tasks/test_commit_context.py:422]: Isn't it better to add a date() here so you make sure in tests the commit is always recent?

File: src/sentry/tasks/commit_context.py [hunk 2]
```diff
@@ -26,6 +33,24 @@
...
 
+def queue_comment_task_if_needed(commit: Commit, group_owner: GroupOwner):
+    from sentry.tasks.integrations.github.pr_comment import comment_workflow
+
+    pr = PullRequest.objects.filter(
+        organization_id=commit.organization_id, merge_commit_sha=commit.key
+    )
+    if (
+        pr.exists()
+        and pr[0].date_added >= datetime.now(tz=timezone.utc) - timedelta(days=30)
```
Comment (sentaur-athena) [inline src/sentry/tasks/commit_context.py:44]: Would pr ever include more than one object? and if so why do we choose pr[0] over the others?

Comment (cathteng) [reply src/sentry/tasks/commit_context.py:44]: `pr` is technically a query so we might want to call it `pr_query`. we could also use `get()` but `get()` throws an error if the item doesn't exist or if multiple items exist that match the criteria

**Review by cathteng — COMMENTED** [review]
looks good to me aside from a few nits and small q's

File: tests/sentry/tasks/test_commit_context.py [hunk 3]
```diff
@@ -408,3 +411,157 @@ def after_return(self, status, retval, task_id, args, kwargs, einfo):
...
+            "commitMessage": "placeholder commit message",
+            "commitAuthorName": "",
+            "commitAuthorEmail": "admin@localhost",
+        }
+    ),
+)
+@patch("sentry.tasks.integrations.github.pr_comment.comment_workflow.delay")
+class TestGHCommentQueuing(TestCommitContextMixin):
```
[the remark this answers is not in the document]
Comment (AniketDas-Tekky) [reply tests/sentry/tasks/test_commit_context.py:430]: yup! Saves on code repetition. I think you can override it too if you have one or two tests that want a different patch

File: tests/sentry/tasks/test_commit_context.py (hunk 1 above)
Comment (AniketDas-Tekky) [reply tests/sentry/tasks/test_commit_context.py:422]: the commit date isn't important in this case. I'd rather just mock a response once here and have the object be fixed.

**Review by cathteng — APPROVED** [review]

**Pull request merged**